src/bio.c

```diff
@@ -66,10 +58,25 @@ static unsigned int bio_job_to_worker[] = {
     [BIO_COMP_RQ_LAZY_FREE]  = 2
 };
 
-static pthread_t bio_threads[BIO_WORKER_NUM];
-static pthread_mutex_t bio_mutex[BIO_WORKER_NUM];
-static pthread_cond_t bio_newjob_cond[BIO_WORKER_NUM];
-static list *bio_jobs[BIO_WORKER_NUM];
+typedef struct {
+    const char *const bio_worker_title;
+    pthread_t bio_thread_id;
+    pthread_mutex_t bio_mutex;
+    pthread_cond_t bio_newjob_cond;
+    list *bio_jobs;
+} bio_worker_data;
+
+static bio_worker_data bio_workers[] = {
+    {"bio_close_file"},
+    {"bio_aof"},
+    {"bio_lazy_free"}
+};
```

Comment:
```suggestion
static bio_worker_data bio_workers[] = {
    {"bio_close_file"},
    {"bio_aof"},
    {"bio_lazy_free"},
    {NULL}
};
```
perhaps we can do like this to avoid introducing `bio_worker_end`.

Reply:
Yes, but why would it be preferable to create a full dummy `bio_worker_data` instance instead of a sentinel?

Reply:
```
static bio_worker_data bio_workers[] = {
    {"bio_close_file"},
    {"bio_aof"},
    {"bio_lazy_free"},
};

#define BIO_WORKER_NUM (sizeof(bio_workers) / sizeof(bio_worker_data))
```

Since I don't want to modify so many lines just for code refactoring, we can try to keep the old way as much as possible and minimize the changes.

Reply:
That would be what the [original patch](https://github.com/redis/redis/commit/0a1205569da3da7130d983ba1f589ac25500e20d) did. I didn't like it myself since that requires every loop to start with `bio_worker_data *const bwd = &bio_workers[j];` so the number of lines changed/added will be very similar. I don't think that added line in every loop adds any value.

Reply:
i prefer the original patch one, which can reduce some changes.

Reply:
@sundb Ok, updated.